haml/_oisc_warning.html.haml

```diff
@@ -0,0 +1,19 @@
+-# OISC Warning
```

Comment:
Correct me if I'm wrong, but it does not look like this PR is only documentation changes? You seem to be committing OISC feature work in this PR too. Can we split that out into it's own PR?

Reply:
> ⚠️ I've included my initial work on OISC warnings in here as a working example. Looking for reviews on the storybook structure at this stage, not this component. If we are happy with the approach I'll remove that work from this PR so we can merge this without shipping early work on the warnings component.